libs/base/Data/List.idr

```diff
@@ -117,6 +119,25 @@ find : (p : a -> Bool) -> (xs : List a) -> Maybe a
 find p [] = Nothing
 find p (x::xs) = if p x then Just x else find p xs
 
+||| Find the index of the first element (if exists) of a list that satisfies
+||| the given test, else `Nothing`.
+public export
+findIndex : (a -> Bool) -> List a -> Maybe Nat
+findIndex p = h 0 where
+  h : Nat -> List a -> Maybe Nat
+  h _ [] = Nothing
+  h lvl (x :: xs) = if p x then Just lvl else h (S lvl) xs
+
+||| Find indices of all elements that satisfy the given test.
+public export
+findIndices : (a -> Bool) -> List a -> List Nat
+findIndices p = h 0 where
+  h : Nat -> List a -> List Nat
+  h _         []  = []
+  h lvl (x :: xs) = if p x
+    then lvl :: h (S lvl) xs
+    else        h (S lvl) xs
+
```

Comment:
I think these better return an informative proof that the natural number is actually an index into the list,
like `InBounds`:
https://github.com/idris-lang/Idris2/blob/0209cee41a2487f42dc81dc9755cab4b87419a10/libs/base/Data/List.idr#L40

Reply:
thx

Reply:
InBounds for snoc too